awssqs/src/main/java/zipkin/reporter/awssqs/AwsBufferedSqsSender.java

```diff
@@ -0,0 +1,235 @@
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import zipkin.internal.LazyCloseable;
+import zipkin.internal.Nullable;
+import zipkin.reporter.Callback;
+import zipkin.reporter.Encoding;
+import zipkin.reporter.Sender;
+
+import static zipkin.internal.Util.checkNotNull;
+
+/**
+ * This sends (usually TBinaryProtocol big-endian) encoded spans to an SQS queue
+ */
+@AutoValue
+public abstract class AwsBufferedSqsSender extends LazyCloseable<AmazonSQSBufferedAsyncClient> implements Sender {
+
+  private static AtomicInteger messageId = new AtomicInteger(0);
+
+  public static AwsBufferedSqsSender create(String url) {
+    return builder().queueUrl(url).build();
+  }
+
+  public static Builder builder() {
+    return new AutoValue_AwsBufferedSqsSender.Builder()
+        .encoding(Encoding.THRIFT)
+        .compressionEnabled(true)
+        .awsSecretAccessKey(null)
+        .awsAccessKeyId(null)
+        .awsRoleName(null)
+        .awsServiceId(null)
+        .maxRequests(64)
+        .maxQueuedRequests(1000)
+        .messageMaxBytes(256 * 1024 - 30); // 256KB SQS limit per message. 30 bytes for attr keys.
+  }
+
+  @AutoValue.Builder
+  public static abstract class Builder {
+
+    public abstract Builder queueUrl(String queueUrl);
+    public abstract Builder awsAccessKeyId(String awsAccessKeyId);
+    public abstract Builder awsSecretAccessKey(String awsSecretAccessKey);
+    public abstract Builder awsRoleName(String awsRoleName);
+    public abstract Builder awsServiceId(String awsServiceId);
+    public abstract Builder compressionEnabled(boolean compressionEnabled);
+    public abstract Builder maxRequests(int maxRequests);
+    public abstract Builder maxQueuedRequests(int maxQueuedRequests);
+
+
+    public abstract Builder encoding(Encoding encoding);
+
+    /**
+     * Maximum size of a message. SQS max message size is 256KB including attributes.
+     * Default 256 less 30 bytes for message keys.
+     *
+     * @param messageMaxBytes
+     * @return
+     */
+    public abstract Builder messageMaxBytes(int messageMaxBytes);
+
+    public final AwsBufferedSqsSender build() {
+      return autoBuild();
+    }
+
+    abstract AwsBufferedSqsSender autoBuild();
+
+    Builder() {}
+  }
+
+  public Builder toBuilder() {
+    return new AutoValue_AwsBufferedSqsSender.Builder(this);
+  }
+
+  abstract String queueUrl();
+  @Nullable abstract String awsAccessKeyId();
+  @Nullable abstract String awsSecretAccessKey();
+  @Nullable abstract String awsRoleName();
+  @Nullable abstract String awsServiceId();
+  abstract boolean compressionEnabled();
+  abstract int maxRequests();
+  abstract int maxQueuedRequests();
+
+  /** close is typically called from a different thread */
+  private AtomicBoolean closeCalled = new AtomicBoolean(false);
+
+
+  @Override public CheckResult check() {
```

Comment:
Agreed.  I was at a loss for a better way to perform health checks.  Requiring extra permissions for this isn't the best.  I will either remove it or make the check optional through config.